indexing-service/src/main/java/io/druid/indexing/overlord/setup/BaseWorkerBehaviorConfig.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord.setup;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.druid.guice.annotations.ExtensionPoint;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = WorkerBehaviorConfig.class)
+@JsonSubTypes(value = {
+    @JsonSubTypes.Type(name = "default", value = WorkerBehaviorConfig.class)
+})
+@ExtensionPoint
+public interface BaseWorkerBehaviorConfig
```

Comment:
aren't names normally the other way around? where something like `X.class` is an interface, and `YX.class` is an implementation or similar?

Should `WorkerBehaviorConfig` and `BaseWorkerBehaviorConfig` switch their names?

Reply:
Aren't there any compatibility concerns with such renaming? Because `WorkerBehaviorConfig` is json-serialized

Reply:
The json serialization should not matter for the actual class name itself, since that is mapped with the value in the `type` field. It will retain the appropriate behavior as long as the `type` key makes it go to the correct implementation.

Reply:
Renamed `BaseWorkerBehaviorConfig` to `WorkerBehaviorConfig` and `WorkerBehaviorConfig` to `DefaultWorkerBehaviorConfig`